### PR TITLE
Added some fixes in sql date to util date conversion for in memory qu…

### PR DESCRIPTION
**Pull request by millevlada**
…eries.

### CODE REVIEW

File: to.etc.domui/src/main/java/to/etc/domui/util/db/CriteriaMatchingVisitor.java [hunk 1]
```diff
@@ -95,6 +96,14 @@ public void visitPropertyComparison(@NonNull QPropertyComparison n) throws Excep
 		Class< ? > litc = DomUtil.getUnproxiedClass(lit.getClass());
 		Class< ? > valc = DomUtil.getUnproxiedClass(val.getClass());	// Types differ?
 		if(litc != valc) {
+			if(valc.isAssignableFrom(java.sql.Date.class) && litc.isAssignableFrom(java.util.Date.class)) {
```
Comment (millevlada) [inline to.etc.domui/src/main/java/to/etc/domui/util/db/CriteriaMatchingVisitor.java:99]: Is this ok, or we like to have some more generic approach?

Comment (fjalvingh) [inline to.etc.domui/src/main/java/to/etc/domui/util/db/CriteriaMatchingVisitor.java:99]: The first thing should be an instanceof, not isAssignableFrom

Comment (fjalvingh) [reply to.etc.domui/src/main/java/to/etc/domui/util/db/CriteriaMatchingVisitor.java:99]: If you look at the code the proper thing to do would be to use the getPromoted method and all. Let me see how bad that would get.

Comment (millevlada) [reply to.etc.domui/src/main/java/to/etc/domui/util/db/CriteriaMatchingVisitor.java:99]: is this now fixed with alternative solution? Then we should close this PR as obsolete.

**Comment by fjalvingh** [comment]
Yes, I have used the existing mechanism to promote both types to the same date type (just added the conversions, and added the promotions to the order list).

**Pull request closed without merging**